src/main/kotlin/no/nav/familie/tilbake/behandling/steg/StegService.kt

```diff
@@ -0,0 +1,59 @@
+package no.nav.familie.tilbake.behandling.steg
+
+import no.nav.familie.tilbake.behandlingskontroll.BehandlingskontrollService
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingssteg
+import no.nav.familie.tilbake.common.exceptionhandler.Feil
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
```

Comment:
Forslag: 
@Service
class StegService(val steg: List<IBehandlingssteg>,
                  val behandlingskontrollService: BehandlingskontrollService) {

    fun håndterSteg(behandlingId: UUID) {
        val aktivtBehandlingssteg = behandlingskontrollService.finnAktivtSteg(behandlingId)
                                    ?: throw  Feil(message = "Behandling $behandlingId har ikke noe aktiv seg",
                                                   frontendFeilmelding = "Behandling $behandlingId har ikke noe aktiv seg")

        if (aktivtBehandlingssteg !in setOf(Behandlingssteg.VARSEL,
                                                   Behandlingssteg.GRUNNLAG,
                                                   Behandlingssteg.FAKTA,
                                                   Behandlingssteg.FORELDELSE,
                                                   Behandlingssteg.VILKÅRSVURDERING)) {
            throw Feil(message = "Støtte for $aktivtBehandlingssteg ikke implementer ennå")
        }

        hentStegInstans(aktivtBehandlingssteg).utførSteg(behandlingId)
    }

    private fun hentStegInstans(behandlingssteg: Behandlingssteg): IBehandlingssteg {
        return steg.singleOrNull { it.getBehandlingssteg() == behandlingssteg }
               ?: error("Finner ikke behandlingssteg $behandlingssteg")
    }
}

Reply:
Nei, det kan ikke gjøres. Koden kaller forskjellige metode basert på aktivtbehandlingssteg

Reply:
Nei, dessverre kan det ikke gjøres. Det gir kompilasjon feil "Not enough information to infer type variable T"